checks/raw/sast.go

```diff
@@ -269,141 +210,11 @@ var searchGitHubActionWorkflowSnyk fileparser.DoWhileTrueOnFileContent = func(pa
-				Path:   path,
-				Offset: checker.OffsetDefault,
-				Type:   finding.FileTypeSource,
-			},
-			Type: checker.PysaWorkflow,
-		}
-
-		sastWorkflows = append(sastWorkflows, sastWorkflow)
-	}
-	return sastWorkflows, nil
-}
-
-var searchGitHubActionWorkflowPysa fileparser.DoWhileTrueOnFileContent = func(path string,
-	content []byte,
-	args ...interface{},
-) (bool, error) {
-	if !fileparser.IsWorkflowFile(path) {
-		return true, nil
-	}
-
-	if len(args) != 1 {
-		return false, fmt.Errorf(
-			"searchPysa requires exactly 1 arguments: %w", errInvalid)
-	}
-
-	// Verify the type of the data.
-	paths, ok := args[0].(*[]string)
-	if !ok {
-		return false, fmt.Errorf(
-			"searchPysa expects arg[0] of type *[]string: %w", errInvalid)
-	}
-
-	workflow, errs := actionlint.Parse(content)
-	if len(errs) > 0 && workflow == nil {
-		return false, fileparser.FormatActionlintError(errs)
-	}
-
-	for _, job := range workflow.Jobs {
-		for _, step := range job.Steps {
-			e, ok := step.Exec.(*actionlint.ExecAction)
-			if !ok || e == nil || e.Uses == nil {
-				continue
-			}
-			// Parse out repo / SHA.
-			uses := strings.TrimPrefix(e.Uses.Value, "actions://")
-			action, _, _ := strings.Cut(uses, "@")
-			if strings.HasPrefix(action, "facebook/pysa-action") {
-				*paths = append(*paths, path)
-			}
-		}
-	}
-	return true, nil
-}
-
-func getQodanaWorkflows(c *checker.CheckRequest) ([]checker.SASTWorkflow, error) {
-	var workflowPaths []string
-	var sastWorkflows []checker.SASTWorkflow
-	err := fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
-		Pattern:       ".github/workflows/*",
-		CaseSensitive: false,
-	}, searchGitHubActionWorkflowQodana, &workflowPaths)
-	if err != nil {
-		return sastWorkflows, err
-	}
-	for _, path := range workflowPaths {
-		sastWorkflow := checker.SASTWorkflow{
-			File: checker.File{
-				Path:   path,
-				Offset: checker.OffsetDefault,
-				Type:   finding.FileTypeSource,
-			},
-			Type: checker.QodanaWorkflow,
-		}
-
-		sastWorkflows = append(sastWorkflows, sastWorkflow)
-	}
-	return sastWorkflows, nil
-}
-
-var searchGitHubActionWorkflowQodana fileparser.DoWhileTrueOnFileContent = func(path string,
-	content []byte,
-	args ...interface{},
-) (bool, error) {
-	if !fileparser.IsWorkflowFile(path) {
-		return true, nil
-	}
-
-	if len(args) != 1 {
-		return false, fmt.Errorf(
-			"searchQodana requires exactly 1 arguments: %w", errInvalid)
-	}
-
-	// Verify the type of the data.
-	paths, ok := args[0].(*[]string)
-	if !ok {
-		return false, fmt.Errorf(
-			"searchQodana expects arg[0] of type *[]string: %w", errInvalid)
-	}
-
-	workflow, errs := actionlint.Parse(content)
-	if len(errs) > 0 && workflow == nil {
-		return false, fileparser.FormatActionlintError(errs)
-	}
-
-	for _, job := range workflow.Jobs {
-		for _, step := range job.Steps {
-			e, ok := step.Exec.(*actionlint.ExecAction)
-			if !ok || e == nil || e.Uses == nil {
+			m, err := regexp.MatchString(useRegex, action)
+			if err != nil {
```

Comment:
Any preference between this and trying to force the pattern to compile? I worry errors from bad regexes here may slip through since you just silently `continue` below. `MustCompile` causes a panic, but this would be a programmer error we'd want to catch.

```go
re := regexp.MustCompile(useRegex)
if re.MatchString(action) {
    *paths = append(*paths, path)
}
```

Reply:
Good point -- I set it to the pattern you suggested.